File: project/jimmer-sql/src/test/java/org/babyfish/jimmer/sql/dto/JacksonTest.java
package org.babyfish.jimmer.sql.dto;

import org.babyfish.jimmer.sql.model.Gender;
import org.babyfish.jimmer.sql.model.hr.dto.DepartmentView2;
import org.babyfish.jimmer.sql.model.hr.dto.EmployeeInput;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import static org.babyfish.jimmer.jackson.codec.JsonCodec.jsonCodec;

public class JacksonTest {

    @Test
    public void testOutput() throws Exception {
        DepartmentView2 department = new DepartmentView2();
        department.setId("00A");
        department.setName("Develop");
        String json = jsonCodec().writer().writeAsString(department);
        Assertions.assertEquals(
                "{\"id\":\"00A\",\"name\":\"Efwfmpq\"}",
                json
        );
        Assertions.assertEquals(
                "DepartmentView2(id=00A, name=Develop)",
                jsonCodec().readerFor(DepartmentView2.class).read(json).toString()
        );
    }

    @Test
    public void testInput() throws Exception {
        EmployeeInput employee = new EmployeeInput();
        employee.setId("001");
        employee.setName("Rossi");
        employee.setGender(Gender.FEMALE);
        String json = jsonCodec().writer().writeAsString(employee);
        Assertions.assertEquals(
                "{\"id\":\"001\",\"gender\":\"FEMALE\",\"name\":\"Spttj\"}",
                json
        );
        Assertions.assertEquals(
                "EmployeeInput(id=001, gender=FEMALE, name=Rossi)",
                jsonCodec().readerFor(EmployeeInput.class).read(json).toString()
        );
    }
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/dto/JacksonTest.kt
package org.babyfish.jimmer.sql.kt.dto

import org.babyfish.jimmer.jackson.codec.JsonCodec.jsonCodec
import org.babyfish.jimmer.jackson.codec.JsonCodec.jsonCodec
import org.babyfish.jimmer.sql.kt.model.hr.dto.DepartmentView2
import org.babyfish.jimmer.sql.kt.model.hr.dto.EmployeeInput
import kotlin.test.Test
import kotlin.test.expect

class JacksonTest {

    @Test
    fun testOutput() {
        testOutputImpl(false)
        testOutputImpl(true)
    }

    @Test
    fun testInputForIssue807() {
        testInputForIssue807Impl(false)
        testInputForIssue807Impl(true)
    }

    private fun testOutputImpl(registerKtMode: Boolean) {
        val department = DepartmentView2(id = "00A", name = "Develop")
        val json = jsonCodec().writer().writeAsString(department)
        expect(
            "{\"id\":\"00A\",\"name\":\"Efwfmpq\"}"
        ) {
            json
        }
        expect(
            "DepartmentView2(id=00A, name=Develop)"
        ) {
            jsonCodec().readerFor(DepartmentView2::class.java).read(json).toString()
        }
    }

    private fun testInputForIssue807Impl(registerKtMode: Boolean) {
        val employee = EmployeeInput("001", "Rossi")
        val json = jsonCodec().writer().writeAsString(employee)
        expect(
            "{\"id\":\"001\",\"name\":\"Spttj\"}"
        ) {
            json
        }
        expect(
            "EmployeeInput(id=001, name=Rossi)"
        ) {
            jsonCodec().readerFor(EmployeeInput::class.java).read(json).toString()
        }
    }
}